tests/components/todo/test_init.py

```diff
@@ -0,0 +1,146 @@
+"""Tests for the todo integration."""
+
+from typing import Any
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.bootstrap import async_setup_component
+from homeassistant.components.todo import DOMAIN
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from tests.typing import WebSocketGenerator
+
+
+@pytest.fixture(autouse=True)
+async def setup_homeassistant(hass: HomeAssistant):
+    """Set up the homeassistant integration."""
+    await async_setup_component(hass, "homeassistant", {})
+
+
+@pytest.fixture(autouse=True)
+async def select_only() -> None:
+    """Enable only the select platform."""
+    with patch(
+        "homeassistant.components.demo.COMPONENTS_WITH_CONFIG_ENTRY_DEMO_PLATFORM",
+        [Platform.TODO],
+    ):
+        yield
+
+
+async def test_list_todo_items(
```

Comment:
We should add tests for the other actions too and the entity state.

Reply:
I have added tests for the state.

The other actions are tested below, but just showing that they are unsupported by demo. When we look at the coverage for todo alone its not great:
```
$ py.test tests/components/todo --cov=homeassistant.components.todo --cov-report=term-missing
Name                                        Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------
homeassistant/components/todo/__init__.py     128     22    83%   85, 175-183, 219-224, 265-270, 304-309
homeassistant/components/todo/const.py         10      0   100%
-------------------------------------------------------------------------
TOTAL                                         138     22    84%
```

However, the actions are also tested by shopping list tests which brings the coverage to 96%:
```
$ py.test tests/components/todo tests/components/shopping_list --cov=homeassistant.components.todo --cov-report=term-missing
Name                                        Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------
homeassistant/components/todo/__init__.py     128      5    96%   85, 180-181, 221-222
homeassistant/components/todo/const.py         10      0   100%
-------------------------------------------------------------------------
TOTAL                                         138      5    96%
```

Where the only failures are when `HomeAssistatError` is raised which doesn't happen (but will with the cloud APIs which we can also exercise)

I can add support to demo but i think it would replicate the functionality in shopping list or local todo -- perhaps you have a different testing approach in mind  (`patch` out the demo entity calls to succeed or raise?).

Given the above, let me know if that is sufficient or if you think there is still additional value we can get from more tests happy to add them if you can elaborate a bit

Reply:
We should not rely on the demo or shopping list tests to cover the todo integration.

Here's how we test the event entity, with a mock platform in at least one test.

https://github.com/home-assistant/core/blob/dev/tests/components/event/test_init.py

Reply:
Thanks, i removed the demo changes and added mock platforms here.